apps/hpc-ftsadmin/src/app/components/flow-form/flow-form.tsx

```diff
@@ -0,0 +1,2295 @@
+    if (!amountOriginalCurrency || !currency) {
+      return null;
+    }
+
+    const filteredChildFlows = shouldExcludeRestricted
+      ? childFlows.filter((childFlow) => !childFlow.restricted)
+      : childFlows;
+
+    const originalAmountDifference = integerToCurrency(
+      currencyToInteger(amountOriginalCurrency) -
+        filteredChildFlows.reduce((acc, cur) => {
+          if (!cur.amountOriginalCurrency) {
+            return acc;
+          }
+          return acc + valueToInteger(cur.amountOriginalCurrency);
+        }, 0)
+    );
+    return `${currency.displayLabel} ${originalAmountDifference}`;
+  };
+
+  const isUsageYearValues = (
+    value: ConsistencyErrorReasons[keyof ConsistencyErrorReasons][number],
+    type: keyof ConsistencyErrorReasons
+  ): value is ConsistencyErrorReasons['usageYears'][number] =>
+    type === 'usageYears';
+  const isDataConsistencyErrorMap = (
+    reason: errors.DataConsistencyErrorReason[number]
+  ): reason is ConsistencyErrorReasonMap => {
+    const KEYS = new Set<string>([
+      'usageYears',
+      'locations',
+      'governingEntities',
+      'organizations',
+    ] satisfies Array<keyof ConsistencyErrorReasons>);
+    return KEYS.has(reason.type);
+  };
+
+  const handleEarmarkingRestriction = (parentValues: FlowLinkProps | null) => {
+    if (!parentValues?.earmarking) {
+      return earmarkingType;
+    }
+    /**
+     * Ordered based on https://fts.unocha.org/glossary
+     */
+    const EARMARKING_FREEDOM = [
+      'Unearmarked',
+      'Softly earmarked',
+      'Earmarked',
+      'Tightly earmarked',
+    ] as const;
+
+    const isParentEarmarking = (
+      earmarking: string
+    ): earmarking is (typeof EARMARKING_FREEDOM)[number] =>
+      new Set<string>(EARMARKING_FREEDOM).has(earmarking);
+
+    const parentEarmarking = parentValues.earmarking.name;
+    if (!isParentEarmarking(parentEarmarking)) {
+      return earmarkingType;
+    }
+
+    const index = EARMARKING_FREEDOM.indexOf(parentEarmarking);
+    const allowedChildEarmarking = new Set<string>(
+      EARMARKING_FREEDOM.slice(index)
+    );
+
+    return earmarkingType.filter((e) =>
+      allowedChildEarmarking.has(e.displayLabel)
+    );
+  };
+  const handleDataConsistencyError = (err: errors.DataConsistencyError) => {
+    const message = err.reason.map((r) => {
+      let messageReason = '';
+      if (!isDataConsistencyErrorMap(r)) {
+        return messageReason;
+      }
+      const { type, values } = r;
+      messageReason += t.t(
+        lang,
+        (s) => s.components.flowForm.submitValidation.dataConsistency,
+        {
+          entity: type,
+          selected: values
+            .map((v) => (isUsageYearValues(v, type) ? v.year : v.name))
+            .join(', '),
+          expected: [
+            ...new Set(
+              values.flatMap((v) =>
+                isUsageYearValues(v, type)
+                  ? v.options.map((o) => `${o}`)
```

Comment:
You don't like `toString()`? Too old school? :P
```suggestion
                  ? v.options.map((o) => o.toString())
```

Reply:
I prefer `${o}` in this case it has no difference but if `o` could have been `null` or `undefined` unintentionally because we typed things wrong, `${o}` would be more robust since `null` or `undefined` would no throw errors.

We could also do 
```ts
v.options.map(String.toString)
```
since `toString()` just takes one parameter is safe to use like so.